Added proposal for handbook.

bwplotka opened the pull request:
Please review 🤗 

Also Preview: https://deploy-preview-12--rhobs-handbook.netlify.app/proposals/accepted/202106-handbook.md/

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

bwplotka commented:
NOTE: Proposal process will be proposed in another PR

dgrisonnet commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+  * [Observability Team Process (Internal)](https://docs.google.com/document/d/1eojXStPdq1hYwv36pjE-vKR1q3dlBbpIx5w_L_v2gNo/edit)
+  * [Monitoring Team Processes (Internal)](https://drive.google.com/drive/folders/1yU5DYhpBLzmp3q9SIhudRMZ1rsmQ_sR2)
+
+> TL;DR: I would like to propose to put all public documentation pieces related to the Observability Group (and not tied to a specific project) in the public GitHub repository called [`handbook`](https://github.com/rhobs/handbook). I propose to review all documents with a similar flow as code and put documents in the form of markdown files that can be read from both GitHub UI and automatically se ...
```
```suggestion
> TL;DR: I would like to propose to put all public documentation pieces related to the Observability Group (and not tied to a specific project) in the public GitHub repository called [`handbook`](https://github.com/rhobs/handbook). I propose to review all documents with a similar flow as code and put documents in the form of markdown files that can be read from both GitHub UI and automatically served on https://rhobs-handbook.netlify.app/ website.
```

dgrisonnet commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+The currently planned audience for proposed documentation content is following (in importance order):
+
+1. Observability Group Team Members (Monitoring Scrum Teams, optionally Logging and Tracing if they choose to follow our process).
+* Also primary writer.
```
I wonder if it is meaningful to mention that here

bwplotka replied:
Yea, maybe does not matter, will delete (:

dgrisonnet commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+
+1. Observability Group Team Members (Monitoring Scrum Teams, optionally Logging and Tracing if they choose to follow our process).
+* Also primary writer.
+1. External Teams and Red Hat.
```
```suggestion
2. External Teams at Red Hat.
```
I am not sure if you really meant `and Red Hat`, but I would expect internal customers to come second here.

dgrisonnet commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+1. Observability Group Team Members (Monitoring Scrum Teams, optionally Logging and Tracing if they choose to follow our process).
+* Also primary writer.
+1. External Teams and Red Hat.
+2. Outside people, contributors to our projects, potential future hires, people interested in best practices, team processes etc.
```
```suggestion
3. Outside people, contributors to our projects, potential future hires, people interested in best practices, team processes etc.
```

dgrisonnet commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+
+## Non-Goals
+
+* Support other formats than `Markdown,` e.g. Asciidoc
```
```suggestion
* Support other formats than `Markdown`, e.g. Asciidoc
```

dgrisonnet commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+
+### Flow of Adding/Consuming Documentation to Handbook
+
+![flow](../../assets/handbook-proposal.png)
```
:100:

fpetkovski commented on README.md:
```diff
@@ -6,4 +6,25 @@
+
+This handbook was designed to be used in two ways:
+
+* Through our website `https://rhobs-handbook.netlify.app/`.
```
It would be nice to use a Red Hat specific domain name instead of a third party one.

ianbillett replied:
IMO the effort required to acquire and serve the relevant certificate is almost certainly not worth the effort :disappointed:

paulfantom replied:
I would say the opposite. Branding this as Red Hat might need some special approvals and having this under "our" domain doesn't need anything like that.

bwplotka replied:
Yea, I would not bother for now, it might be not very important for us. (:

fpetkovski replied:
The handbook is called Red Hat Observability Group Handbook, which to me already seems branded as Red Hat. 
I understand the limitations of obtaining a RH domain name, but I still netlify.app a bit unprofessional. Based on this proposal, the handbook will be accessible and used as a reference outside of the RHOBS group. 

As an alternative I would suggest using Github pages.

bwplotka replied:
Thanks, I see your point but:

* We can apply for a more branded website, there is a process we can start anytime. I don't see a blocker, but it will take time.
* What gives us more professionalism? Previously there was no documentation and only half baked Google Docs so I consider this a major improvement (: As the audience states we want primarily to have our team offloaded from knowledge repetitive in-meetings knowledge sharing by having our handbook. I don't think for our team it matters what domain it is. For external Red Hat teams too. For customers, maybe but they will rarely visit this page in current proposal.
* What's the difference between Netlify vs GitHub?

squat replied:
> What's the difference between Netlify vs GitHub?

Slightly nicer URLs that are more like vanity URLs and a more globally recognized name that inspires some confidence. Otherwise not much really.

fpetkovski replied:
If we are not going to use the domain outside of RH, I agree that it does not really matter much which one we use. 

> What's the difference between Netlify vs GitHub?

Netlify is yet another tool to learn, manage and request access for. With GH pages the user management model is tied to the repository. I also assume the Netlify account is a private one which makes things complicated when people leave the team. 

In any case, not a blocker in any way for the proposal, just wanted to raise some potential pitfalls.

bwplotka replied:
Yes, thanks! We have the netlify heavly used and relied on for years already on other project (Thanos, Observatorium, Prometheus,)(, that's why this choice. Once website will be more active we will ask for an open source tier which gives us team permissions. Added:

* https://github.com/rhobs/handbook/issues/14
* https://github.com/rhobs/handbook/issues/13

fpetkovski commented on README.md:
```diff
@@ -6,4 +6,25 @@
+This handbook was designed to be used in two ways:
+
+* Through our website `https://rhobs-handbook.netlify.app/`.
+* Directly from GitHub UI by navigating to [`content`](https://github.com/rhobs/handbook/tree/main/content) directory.
```
What about Github pages?

bwplotka replied:
Do you mean instead of hugo + netlify? We have some amazing pre-processors which makes GitHub UI and website working smoothly (you write docs like for GH UI) (: GH pages would lack of that.

Also we have rdy tooling as other website use it, so we can maintain one way of generating website.

fpetkovski replied:
> We have some amazing pre-processors which makes GitHub UI and website working smoothly

Interesting, could you send me some links?

bwplotka replied:
* https://github.com/bwplotka/mdox/blob/main/main.go#L206
* https://github.com/rhobs/handbook/blob/f1860c7564b0f84192c26302aebfb3c3bbd49fc8/.mdox.yaml#L11

fpetkovski commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+  * [Observability Team Process (Internal)](https://docs.google.com/document/d/1eojXStPdq1hYwv36pjE-vKR1q3dlBbpIx5w_L_v2gNo/edit)
+  * [Monitoring Team Processes (Internal)](https://drive.google.com/drive/folders/1yU5DYhpBLzmp3q9SIhudRMZ1rsmQ_sR2)
+
+> TL;DR: I would like to propose to put all public documentation pieces related to the Observability Group (and not tied to a specific project) in the public GitHub repository called [`handbook`](https://github.com/rhobs/handbook). I propose to review all documents with a similar flow as code and put documents in the form of markdown files that can be read from both GitHub UI and automatically se ...
```
This makes complete sense. I would just make it more explicit that developer docs (things like how to build, test and run a project) should be kept in the repository of the project itself.

bwplotka replied:
Yes, it's part of the flow below, but let me add explicitly here too.

fpetkovski commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+* Harder to share with external teams
+* We can't use open source tools, CIs etc.
+
+## Action Plan
```
It would be great to have a timeline until when the docs would be moved if the proposal is approved. Should this be included in the roadmap for Q3?

In addition to that, do we know where to find existing documentation? Does it only exist in google drive so far, and what is the exact content that needs to be moved to the handbook?

bwplotka replied:
Good point, let me add this info:

* when
* where to find now
* how to extract

jan--f reviewed (approved):
Love it!

One thing that could be mentioned imho is our user documentation (i.e https://docs.openshift.com/container-platform/4.7/monitoring/understanding-the-monitoring-stack.html). I think we should either explicitly mention it as a non-goal (since I suppose we have processes for that) or mention it as a future item.

dgrisonnet reviewed (commented):
This is an awesome initiative @bwplotka, thank you for looking into that :slightly_smiling_face:

I have a couple of questions that you might be able to ask in regards to that initiative:

- How I see one part of the project is as a central place that leverages upstream documentation or any other learning resources and supplements it. One thing that might be hard to understand today is the actual boundary between what should be contributed in the handbook and what should be contributed upstream. Do you think it would be meaningful to document that?
-  What's the process for upstream enhancements? Should we start drafts in this repository before submitting them to the upstream communities?

fpetkovski commented on content/contributing.md:
```diff
@@ -1,3 +1,105 @@
+
+When you run `make web` or `make web-server` a particular process is happening to "render" the website:
+
+1. `mdox` using `.mdox.yaml` is transforming the resources from `content` directory, moving into website friendly form (adding frontmatter, adjusting links and many others) and outputs results in `./web/content` and `./web/static` which are never committed (temporary directories).
```
Is there an actual need for `mdox`? Could we simply stick with mdbook, mkdocs or some other widely used tool for static page generation?

bwplotka replied:
I kind of set ambitous goal to make sure you don't worry about website - you just write markdown like you would want to have it exposed on GH UI, mdox will do rest for you. We are growing this project as default website preprocess/tooling with @saswatamcode 

If this become a bottleneck we can switch to just hugo with magic fontmatters and directives, but let's see what the experience is maybe?

fpetkovski replied:
Ok, let's give it a shot :+1:

kakkoyun reviewed (approved):
:heart: LGTM

paulfantom commented:
> What's the process for upstream enhancements? Should we start drafts in this repository before submitting them to the upstream communities?

I would vote "upstream first" as docs have this nasty tendency of being stale and not updated or propagated. We can always use links and reference issues/PRs/other pages, no?

ianbillett commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+The currently planned audience for proposed documentation content is following (in importance order):
+
+1. Observability Group Team Members (Monitoring Scrum Teams, optionally Logging and Tracing if they choose to follow our process).
+* Also primary writer.
```
```suggestion
    * Also primary writer.
```
missing indents

paulfantom commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,133 @@
+## Non-Goals
+
+* Support other formats than `Markdown,` e.g. Asciidoc
+* Precise design proposal process (it will come in a separate proposal).
```
Can we align it with Kubernetes Enhancement Proposal process? This way we would have a sort of test bed for writing KEPs :)

bwplotka replied:
Yes, that's the plan, just wanted to keep website and proposals idea separate (although the process will be similar)

dgrisonnet replied:
Just to mention the tooling behind KEPs is https://github.com/kubernetes-sigs/mdtoc, and they use the following scripts to verify/update the format of each KEP:
- https://github.com/thockin/k8s-enhancements/blob/cccc6fa97d5d23e02ad4530d660d306cf49d8c35/hack/update-toc.sh
- https://github.com/thockin/k8s-enhancements/blob/cccc6fa97d5d23e02ad4530d660d306cf49d8c35/hack/verify-toc.sh

bwplotka replied:
oh, super useful, will use in next proposal

bwplotka replied:
hm... although it "only" generates TOC, does not verify any format,but cool idea.

ianbillett reviewed (approved):
I really like that we are aspiring to follow GitLab's examples of open-first documentation :heart: 

A shiny new website **alone** is not going to solve problems we experience when working with documentation.

Let's try it out :muscle:

bwplotka reviewed (commented):
Thanks, everyone for the quick review!

 > One thing that could be mentioned imho is our user documentation (i.e https://docs.openshift.com/container-platform/4.7/monitoring/understanding-the-monitoring-stack.html). I think we should either explicitly mention it as a non-goal (since I suppose we have processes for that) or mention it as a future item.

Yes good point @jan--f 

>  How I see one part of the project is as a central place that leverages upstream documentation or any other learning resources and supplements it. One thing that might be hard to understand today is the actual boundary between what should be contributed in the handbook and what should be contributed upstream. Do you think it would be meaningful to document that?

I added the handbook-proposal.png with the algorithm to decide. Are there any other rules we can add to make sure it's clearer? (: 

> What's the process for upstream enhancements? Should we start drafts in this repository before submitting them to the upstream communities?
>  would vote "upstream first" as docs have this nasty tendency of being stale and not updated or propagated. We can always use links and reference issues/PRs/other pages, no?

Design doc to be added soon in next PR, but the plan is similar to this proposal really. (similar to KEP)

@dgrisonnet

bwplotka commented:
Thanks I  think I addressed all comments, PTAL

sthaha commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,136 @@
+# 2021-06: Handbook
+
+* **Owners:**:
```
nit: extra `:`

sthaha commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,136 @@
+* Without a place for written design proposals (those in progress, those accepted and rejected), the team risks repeating iterating over the same ideas or challenging old ideas already researched.
+* Without good operational or configuration knowledge, we keep asking the same question about, e.g. how to rollout service X or contribute to X etc.
+
+##### Despite strong incentives, writing documentation was proven to be one the most unwanted task among engineers
```
```suggestion
##### Despite strong incentives, writing documentation was proven to be one of the most unwanted task among engineers
```

bwplotka commented:
👋🏽 I changed: 
* Moved naming of this handbook to `Monitoring Group` to reflect that we have only monitoring team in the process nowhere.
* Articulated clearly, that it's not to replace any product documentation - it's mainly for our own documentation to not get lost, have smooth guides and index for further resources.

bwplotka commented:
If no objection, will merge in couple of hours/Monday. Anyone want to take a look still?

sthaha commented:
/approve

kyoto commented on content/Proposals/Accepted/202106-handbook.md:
```diff
@@ -0,0 +1,139 @@
+* Without a place for written design proposals (those in progress, those accepted and rejected), the team risks repeating iterating over the same ideas or challenging old ideas already researched.
+* Without good operational or configuration knowledge, we keep asking the same question about, e.g. how to rollout service X or contribute to X etc.
+
+##### Despite strong incentives, writing documentation was proven to be of one the most unwanted task among engineers
```
```suggestion
##### Despite strong incentives, writing documentation has proven to be of one the most unwanted task among engineers
```

kyoto commented:
LGTM
